src/command_modules/azure-cli-role/azure/cli/command_modules/role/commands.py

```diff
@@ -12,49 +12,108 @@
-                           factory)
-
-factory = lambda _: _auth_client_factory().role_assignments
-cli_command(__name__, 'role assignment delete', 'azure.cli.command_modules.role.custom#delete_role_assignments')
-cli_command(__name__, 'role assignment list', 'azure.cli.command_modules.role.custom#list_role_assignments', table_transformer=transform_assignment_list)
-cli_command(__name__, 'role assignment create', 'azure.cli.command_modules.role.custom#create_role_assignment')
-
-factory = lambda _: _graph_client_factory().applications
-cli_command(__name__, 'ad app create', 'azure.cli.command_modules.role.custom#create_application', factory)
-cli_command(__name__, 'ad app delete', 'azure.cli.command_modules.role.custom#delete_application', factory)
-cli_command(__name__, 'ad app list', 'azure.cli.command_modules.role.custom#list_apps', factory)
-cli_command(__name__, 'ad app show', 'azure.cli.command_modules.role.custom#show_application', factory, exception_handler=empty_on_404)
-cli_command(__name__, 'ad app update', 'azure.cli.command_modules.role.custom#update_application', factory)
-
-factory = lambda _: _graph_client_factory().service_principals
-cli_command(__name__, 'ad sp create', 'azure.cli.command_modules.role.custom#create_service_principal')
-cli_command(__name__, 'ad sp delete', 'azure.cli.command_modules.role.custom#delete_service_principal', factory)
-cli_command(__name__, 'ad sp list', 'azure.cli.command_modules.role.custom#list_sps', factory)
-cli_command(__name__, 'ad sp show', 'azure.cli.command_modules.role.custom#show_service_principal', factory, exception_handler=empty_on_404)
-#RBAC related
-cli_command(__name__, 'ad sp create-for-rbac', 'azure.cli.command_modules.role.custom#create_service_principal_for_rbac')
-cli_command(__name__, 'ad sp reset-credentials', 'azure.cli.command_modules.role.custom#reset_service_principal_credential')
-
-factory = lambda _: _graph_client_factory().users
-cli_command(__name__, 'ad user delete', 'azure.graphrbac.operations.users_operations#UsersOperations.delete', factory)
-cli_command(__name__, 'ad user show', 'azure.graphrbac.operations.users_operations#UsersOperations.get', factory, exception_handler=empty_on_404)
-cli_command(__name__, 'ad user list', 'azure.cli.command_modules.role.custom#list_users', factory)
-cli_command(__name__, 'ad user create', 'azure.cli.command_modules.role.custom#create_user', factory)
-
-factory = lambda _: _graph_client_factory().groups
-cli_command(__name__, 'ad group delete', 'azure.graphrbac.operations.groups_operations#GroupsOperations.delete', factory)
-cli_command(__name__, 'ad group show', 'azure.graphrbac.operations.groups_operations#GroupsOperations.get', factory, exception_handler=empty_on_404)
-cli_command(__name__, 'ad group list', 'azure.cli.command_modules.role.custom#list_groups', factory)
+                           get_role_definition_op('get'),
+                           get_role_definition_op('create_or_update'),
```

Comment:
`get_role_definition_op()` is only used twice and the usage is right after each other.
Not sure why this one deserves a function but the other ones don't.
I'm fine with leaving as is though.